data/data/openstack/masters/main.tf

```diff
@@ -40,21 +40,84 @@ resource "openstack_compute_servergroup_v2" "master_group" {
   policies = ["soft-anti-affinity"]
 }
 
-resource "openstack_compute_instance_v2" "master_conf" {
-  name = "${var.cluster_id}-master-${count.index}"
-  count = var.instance_count
+# The master servers are created in three separate resource definition blocks,
+# rather than a single block with a "count" meta-property, because they need to
+# be created sequentially rather than concurrently by Terraform.
+#
+# The reason why they need to be created one at a time is that OpenStack's
+# Compute module is currently unable to honour the "soft-anti-affinity" policy
+# when the servers are created concurrently.
+#
+# We chose to unroll the loop into three instances, because three is the
+# minimum number of required Control plane nodes, as stated in the
+# documentation[1].
+#
+# The expectation is that machine-api-operator will take care of creating any
```

Comment:
The machine-api-operator is not involved in creating control plane nodes. Adding or removing control plane nodes after installation requires manual intervention by the user. However, there are plans to change this in the future.

Reply:
cf. https://issues.redhat.com/browse/OSASINFRA-2262